### ISSUE TITLE
Automatically passing Consent Request for Precompiled Assemblies 

### ISSUE THREAD

**Issue by Enhar-Turkova** [googleads/googleads-mobile-unity#3189]
### [REQUIRED] Step 1: Describe your environment

  * Unity version: 2022.3.19f1 lts
  * Google Mobile Ads Unity plugin version: _____
  * Platform: (iOS, Android, Unity Editor)
  * Platform OS version: _____ (eg iOS 10, Android 9)
  * Any specific devices issue occurs on: _____
  * Mediation ad networks used, and their versions: _____

### [REQUIRED] Step 2: Describe the problem
Hi everyone, I am trying to build automation for our project. I am doing it by using the CI/CD pipeline and utilizing gitlab. My problem is when I do that, gitlab always clones projects, and when the project cloned consent pop-up shows up on first opening, I can't pass that pop-up without clicking "yes." But for full automation, I have to pass that pop-up automatically. Is there any workaround that you can suggest?

**Comment by mertcelen** [comment]
+1 looking for solution for this problem.

**Comment by NVentimiglia** [comment]
@Enhar-Turkova 

> My problem is when I do that, gitlab always clones projects, and when the project cloned consent pop-up shows up on first opening

Please confirm my assumption:  From the sound of it because the project is cloned previous "yes" clicks are not recorded and forces the automation into the UMP flow. Is this correct? 

>  I have to pass that pop-up automatically. Is there any workaround that you can suggest?

I am not sure what you mean by this. I have not used the Github CI/CD pipeline in gitlab; Can you tell it to select the yes button, or what are you trying to test?

**Comment by Enhar-Turkova** [comment]
I attached the popup below When I send a function from gitlab it does every next and previous steps.Without me having to touch the keyboard just because of this pop up my proccess is not fully automate.Can you suggest me any solution? Btw when I clicked yes and push the library and everything by deleting .gitignore it worked but as you know it is not a valid solution.This just helps me understand that the system is fully working other than this popup

![Screenshot 2024-02-23 at 14 44 35](https://github.com/googleads/googleads-mobile-unity/assets/158572854/3dbaac6b-ef07-41bd-85be-0174254e0ada)

**Comment by NVentimiglia** [comment]
@Enhar-Turkova 

The Unity Consent Request is a feature of the Unity Engine. We can not control this dialog.

I spent some time looking for a solution online, including the [command line arguments](https://docs.unity3d.com/Manual/EditorCommandLineArguments.html), and was unable to find any resources. I did reset my Unity project directory and reinstalled the plugin and the warning did not appear for me (Unity 2023). So either this feature is no longer present on newer version of Unity or the consent status is cached somewhere outside of the project folder.
